sandbox/modules/identity/src/main/java/org/opensearch/identity/rest/action/permission/put/RestPutPermissionAction.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.identity.rest.action.permission.put;
+
+import com.fasterxml.jackson.core.JsonParseException;
+import com.fasterxml.jackson.databind.JsonNode;
+import org.opensearch.client.node.NodeClient;
+import org.opensearch.identity.DefaultObjectMapper;
+import org.opensearch.identity.rest.RestConstants;
+import org.opensearch.identity.utils.ErrorType;
+import org.opensearch.rest.BaseRestHandler;
+import org.opensearch.rest.RestRequest;
+import org.opensearch.rest.action.RestStatusToXContentListener;
+
+import java.io.IOException;
+import java.util.List;
+
+import static java.util.Arrays.asList;
+import static org.opensearch.identity.utils.RoutesHelper.addRoutesPrefix;
+import static org.opensearch.rest.RestRequest.Method.PUT;
+
+/**
+ * Rest action for adding a permission to the permission store
+ */
+public class RestPutPermissionAction extends BaseRestHandler {
+
+    /**
+     * @return a string of the action name  -- "_identity/api/permissions/put
+     */
+    @Override
+    public String getName() {
+        // permission_action_put
+        return RestConstants.PERMISSION_ACTION_PREFIX + "_put";
+    }
+
+    /**
+     * Rest request handler for granting a permission to a subject
+     * @param request the request to execute
+     * @param client  client for executing actions on the local node
+     * @return the action to be executed See {#handleRequest(RestRequest, RestChannel, NodeClient) for more}
+     * @throws IOException if errors encountered when parsing from XContent
+     *
+     * ````
+     * Sample Request:
+     * curl -XPUT http://new-user:password@localhost:9200/_identity/api/permissions/second_user/put --data '{ "permissionString" : "my_permission" }' -H"Content-type: application/json"
+     *
+     *
+     * Sample Response
+     *
+     * {
+     *   "permissions": [
+     *     {
+     *       "successful": true,
+     *       "permission added": "my_permission",
+     *       "username": "second_user"
+     *     }
+     *   ]
+     * }
+     *
+     */
+    @Override
+    @SuppressWarnings("unchecked")
+    public RestChannelConsumer prepareRequest(RestRequest request, NodeClient client) throws IOException {
+        String username = request.param("username");
+
+        // Parsing request body using DefaultObjectMapper
+        JsonNode contentAsNode;
+        try {
```

Comment:
I see this in @DarshitChanpura's PR too and the sec plugin. We should either have a general solution for this for all endpoints introduced or look into how core does this. This is an example I am currently looking at to understand how core unpacks the body of a request - https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/rest/action/admin/cluster/RestPutStoredScriptAction.java#L82-L84

Reply:
Peter felt that using Xcontent directly was messy. DC implemented the ObjectMapper solution so that we would be able to avoid it. Core actually has instances of both use types.

Reply:
fromXContent(..) or in this example parse(..) is a little messy and complicated. However, DefaultObjectMapper uses SecurityManager to parse the object, and it is marked for deprecation. We may need to find an alternative for that